src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/ProxyWhitelist.java

```diff
@@ -113,6 +138,12 @@ public final void reset(Collection<? extends Whitelist> delegates) {
             for (ProxyWhitelist pw : wrappers.keySet()) {
                 pw.reset();
             }
+            if (this.wrappers.isEmpty()) {  // Top-level ProxyWhitelist should be the only cache
+                // Top-level ProxyWhitelist should precache the statically permitted signatures
+                ((EnumeratingWhitelist)(this.delegates.get(0))).precache();
```

Comment:
More easily, make the cache be done in `EnumeratingWhitelist.<init>` (as above), and defer the creation of this until now:

```java
this.delegates.add(0, new EnumeratingWhitelist() {…});
```

Reply:
Enumerating whitelists will likely be created before a ProxyWhitelist wrapping them -- no sense in forcing full cache initialization when it'll just get blown away again.

Reply:
Yeah, though cache generation is cheap and just happens a few times during startup.

At any rate, could be done easily if desired by just having the cache field be nullable—initialize on demand.